### ISSUE TITLE
🐛 Parse error for JSONC files with comments in .vscode/settings.json and .cursor/settings.json

### ISSUE THREAD

**Issue by PaulRBerg** [biomejs/biome#8451]
### Environment information

```
macOS Tahoe 26.1
```

### What happened?

Biome incorrectly reports a parse error when linting `.vscode/settings.json` files that contain comments:

```
.vscode/settings.json:8:3 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ expected `,` but instead found `"tailwindCSS.validate"`
  
     6 │   "tailwindCSS.classFunctions": ["cva", "cn", "tv", "tw", "tw.*"],
     7 │   "tailwindCSS.lint.suggestCanonicalClasses": "ignore", // disabled because it conflicts with eslint-plugin-better-tailwindcss
   > 8 │   "tailwindCSS.validate": true,
       │   ^^^^^^^^^^^^^^^^^^^^^^
     9 │   "typescript.experimental.useTsgo": true,
    10 │   "editor.codeActionsOnSave": {
  
  ℹ Remove "tailwindCSS.validate"
```

The problem is that VS Code's `settings.json` uses JSONC (JSON with Comments), which allows `//` style comments. Biome appears to be parsing these files as strict JSON instead of JSONC.

This also applies to `.cursor/settings.json`, which uses the same JSONC format.

### Expected result

Biome should recognize `.vscode/settings.json` and `.cursor/settings.json` as JSONC files and allow comments in them.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
It's not documented, but VSCode settings file should be parsed using JSONC, so I ask you two things:
- provide the environment information as we requested (biome rage)
- a minimal reproduction 

As for cursor, it's not part of known files. It's not a bug, but a feature. Feel free to send a PR to add it.

**Comment by PaulRBerg** [comment]
## Reproduction

https://github.com/PaulRBerg/biome-repro-1765969535266

## Findings

The issue is that `overrides` from parent configs are not being inherited when the child config specifies `files.includes`.

### Setup

1. Extend from `ultracite/core` which has `overrides` with `allowComments: true` for `*.jsonc` files
2. Add a local `files.includes` to filter which files to check:

```jsonc
{
  "extends": ["ultracite/core"],
  "files": {
    "includes": ["**/*.{json,jsonc}"]
  }
}
```

### Expected Behavior

Parent's `overrides` should still be inherited since the child config doesn't touch `overrides`. These are independent config keys.

### Actual Behavior

JSONC parsing breaks — Biome treats `*.jsonc` files as strict JSON, failing on comments:

```
.vscode/settings.jsonc:3:57 parse
  ✖ Expected a property but instead found '// disabled because...'.
```

### Workaround

Re-declare the `overrides` in the child config:

```jsonc
{
  "extends": ["ultracite/core"],
  "files": {
    "includes": ["**/*.{json,jsonc}"]
  },
  "overrides": [
    {
      "includes": ["**/*.jsonc"],
      "json": {
        "parser": {
          "allowComments": true,
          "allowTrailingCommas": true
        }
      }
    }
  ]
}
```

This suggests a bug in config merging where setting `files` causes `overrides` inheritance to break.

**Comment by ematipico** [comment]
Can you please reduce the reproduction even further without the use of external extends? You can use local extends, so that the debugging is easier

**Comment by PaulRBerg** [comment]
I've updated the repo so that it only uses local extends: https://github.com/PaulRBerg/biome-repro-1765969535266/commit/e4d2ef69beab950ac495f064f178045d83a621c6

**Comment by ematipico** [comment]
This is caused by ultracite. They use a configuration that causes this issue. 

They first disable comments for all `json` and `jsonc` files:

```json
{
  "json": {
    "parser": {
      "allowComments": false,
      "allowTrailingCommas": false
    }
  }
}
```

This also causes issues where all JSONC files can't support comments anymore.

Then, the final override is botched. There's a `"**/.vscode/*.json",`, but then the reproduction has a `jsonc` file, which inherits the incorrect default.

Please file an issue in their repository.

**Comment by PaulRBerg** [comment]
Oh. Sorry for the false positive. Thanks for tracking this down @ematipico
